File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TripKitConfigs.java
package com.skedgo.tripkit;

import org.immutables.value.Value;

@Value.Immutable
public abstract class TripKitConfigs implements Configs {
    public static ImmutableTripKitConfigs.Builder builder() {
        return ImmutableTripKitConfigs.builder();
    }

    @Value.Default
    public boolean debuggable() {
        return false;
    }

    @Value.Default
    public boolean isUuidOptedOut() {
        return false;
    }

    @Value.Default
    public boolean hideTripMetrics() {
        return false;
    }

    @Value.Default
    public boolean showReportProblemOnTripAction() {
        return false;
    }

}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TripKitConfigs.kt
package com.skedgo.tripkit

import org.immutables.value.Value
import org.immutables.value.Value.Default
import org.immutables.value.Value.Immutable

@Immutable
abstract class TripKitConfigs : Configs {
    companion object {
        fun builder(): ImmutableTripKitConfigs.Builder {
            return ImmutableTripKitConfigs.builder()
        }
    }

    @Default
    override fun debuggable(): Boolean {
        return false
    }

    @Default
    override fun isUuidOptedOut(): Boolean {
        return false
    }

    @Default
    override fun hideTripMetrics(): Boolean {
        return false
    }

    @Default
    override fun showReportProblemOnTripAction(): Boolean {
        return false
    }

    @Default
    override fun showOperatorNames(): Boolean {
        return false
    }
}
